pulp_ansible/tests/functional/cli/test_role_import.py

```diff
@@ -0,0 +1,97 @@
+"""Tests that Roles hosted by Pulp can be installed by ansible-galaxy."""
+
+import subprocess
+
+from pulp_smash.pulp3.bindings import monitor_task
+
+from pulp_ansible.tests.functional.utils import TestCaseUsingBindings
+from pulp_ansible.tests.functional.constants import (
+    ANSIBLE_DOCKER_ROLE_NAMESPACE,
+    ANSIBLE_DOCKER_ROLE_NAME,
+    ANSIBLE_DOCKER_ROLE_REPO,
+)
+from pulp_ansible.tests.functional.utils import set_up_module as setUpModule  # noqa:F401
+
+
+class ImportRoleTestCase(TestCaseUsingBindings):
```

Comment:
I'd be ok to accept this style for now as the other tests also have it. It would be much nicer as pytest, but if that's time-prohibitive in this PR, I understand.

Reply:
Can you point me at an example functional test here or elsewhere that meets the criteria? I thought TestCaseUsingBindings was what you were steering me towards yesterday, so I must have misunderstood the expectation.

Reply:
Sure! This [PR shows](https://github.com/pulp/pulp_ansible/pull/1069/files) a rewrite similar to what I'm talking about. It has these kinds of qualities:

* It's pytest native, so it no longer inherits from unittest at all (`TestCaseUsingBindings` inherits from `unittest`).
* It uses pytest fixtures for it's arrangement and cleanup. You can see the pytest fixtures for  [pulp_ansible here](https://github.com/pulp/pulp_ansible/blob/main/pulp_ansible/tests/functional/conftest.py), [pulpcore here](https://github.com/pulp/pulpcore/blob/main/pulpcore/tests/functional/__init__.py) and [pulp-smash](https://github.com/pulp/pulp-smash/blob/main/pulp_smash/pulp3/pytest_plugin/__init__.py) provides some too. `pytest --fixtures` will show you the full list.
* It's bindings based. Your code here is too, but receive the bindings objects from pytest fixtures.
* Use native `assert` instead of self.AssertEqual, etc
* Generate any data you need at runtime, so you can run a test over and over without every having to call orphan cleanup. For this we use [`build_collection` from oriontools](https://github.com/pulp/pulp_ansible/blob/main/pulp_ansible/tests/functional/conftest.py#L131).
* It's ok to leave behind Artifacts (since they are unique per test run) and any kind of Content (also unique per test run).
* Have no tests depend on other tests.
* Mark any test that can be run in parallel with `@pytest.mark.parallel` and any test that should run on in the nightly as `@pytest.mark.nightly`.

If you grep for `pytest` through the test folder you'll find many already, and it likely has examples of a lot of the parts you want to do here.

Reply:
I think this is fixed now, although maybe a bit messy.